docs/cluster_api.md

```diff
@@ -0,0 +1,121 @@
+# Cluster Operator API
+
+## Overview
+
+The cluster operator API defines a set of resources that allow the instantiation of OpenShift clusters and subsequent
+updates to those clusters based on changes to said resources.
+
+The general approach for these resources is to follow the Kubernetes model of a spec section that is end-user controlled and
+a status section that represents the current state of the resources. Changes to the spec will result in the necessary actions
+by the system to reconcile the current state of the system with what the user has specified.
+
+## Resources
+
+
+The cluster resource should allow 2 levels/scopes of spec:
+- Cluster-wide configuration
+- NodeGroup configuration
+
+Furthermore, each scope can be described in 2 parts:
+- Hardware spec (needed to instantiate the proper virtual hardware)
+- OpenShift config (openshift-specific configuration)
+
+For the Status part, if the cluster has been instantiated, we need to know how to access it (Kubeconfig may be sufficient)
+
+In more detail:
+
+### Cluster
+
+- `ObjectMetadata (inline) [metav1.ObjectMetadata]`
+- `TypeMetadata (inline) [metav1.TypeMetadata]`
+
+#### Spec:
+- `Hardware`
+  - `AWS: [struct, optional]` *AWS-specific hardware configuration for the cluster*
+    - `AccountSecret [corev1.LocalReference]` *Reference to a secret that contains an AWS access key id and secret access key*
+    - `SSHKeyName [string]` *SSH secret name to use for machines in this cluster (ie. 'libra')*
+    - `SSHSecret [corev1.LocalReference]` *Reference to a secret that contains a SSH private key to SSH into the cluster*
+    - `Region [string]` *Region where the cluster will be created (us-east-1 for example)*
+    - `VPCName [string, optional]` *Name of the VPC to use for this cluster (or create if not present)*
+    - `VPCSubnet [string]` *Subnet of the VPC (only used if creating a new one)*
+- OpenShift Configuration: *Configuration of the cluster that's not cloud-provider dependent*
```

Comment:
I'm not seeing a place here to define the actual cluster name. Am I just missing it?

We need it in order to have multiple kube clusters in the same AZ: [KubernetesCluster](https://github.com/kubernetes/kubernetes/blob/master/pkg/cloudprovider/providers/aws/tags.go#L30-L34).